backend/testfiles/execution/json.tests

```diff
@@ -315,6 +316,33 @@ Json.parse<Person> """{ "Name": "Bob", "Age": 42, "Height": "6 ft" }""" = Ok (Pe
  | Error _ -> 0) = 42
 
 
+// #### Aliases
+type MyInt = Int
+
+Json.serialize<MyInt> 42 = Ok "42"
+Json.parse<MyInt> "42" = Ok 42
+
+type MyPerson= Person
+Json.serialize<MyPerson> (Person { Name = "Bob"; Age = 42 }) = Ok """{"Age":42,"Name":"Bob"}"""
+Json.parse<MyPerson> """{ "Name": "Bob", "Age": 42 }""" = Ok (Person { Name = "Bob"; Age = 42 })
+type MyPrettyLikely = PrettyLikely
+Json.serialize<MyPrettyLikely> (PrettyLikely.Enh("printer broke", 7)) = Ok "{\"Enh\":[\"printer broke\",7]}"
+Json.parse<MyPrettyLikely> "{\"Enh\":[\"printer broke\",7]}" = Ok (PrettyLikely.Enh("printer broke", 7))
+
+
+type Person1 = { Name: String; Age: Int }
+type Person2 = { Name: String; Age: Int }
+
+Json.serialize<Person2> (Person1 { Name = "Bob"; Age = 42 }) = Error "Incorrect record type"
+
+type StringResult<’a> = Result<’a, String>
+Json.serialize<StringResult<Int>> (Ok 1) = Ok "{\"Ok\":1}"
+Json.serialize<StringResult<Int>> (Error “err”) = Ok "{\"Error\":\"err\"}"
+Json.serialize<StringResult<Int>> (Error 1) = Error "Incorrect type for Error value"
+
+// TODO: This test should fail but doesn't
+// Json.serialize<StringResult<Int>> (Error 1) = Ok "{\"Error\":1}"
```

Comment:
Should we follow-up on this in a different PR?

Reply:
Yeah - respecting type args feels out of scope for this PR, though we do need to follow up on that before too long.